spyder/plugins/ipythonconsole/plugin.py

```diff
@@ -425,11 +437,10 @@ def _remove_old_std_files(self):
                     except Exception:
                         pass
 
-    def _get_working_directory(self):
-        """Get current working directory from its plugin."""
-        workdir = self.get_plugin(Plugins.WorkingDirectory)
-        if workdir:
-            return workdir.get_workdir()
+    @Slot(str)
+    def _save_working_directory(self, new_dir):
+        """Save current working directory on the main widget."""
+        self.get_widget().current_working_directory = new_dir

```

Comment:
I think it will be best to have a function inside the container that sets the current working directory rather than accessing attributes directly

Reply:
Ok, I'll do the necessary changes.